### ISSUE TITLE
numpad not working in neovim

### ISSUE THREAD

**Issue by HalanoSiblee** [summaryInfo/nsst#2]
I've a problem using numbers from numpad in neovim !
![image](https://github.com/user-attachments/assets/c0531165-6c17-4da8-ae74-528b4f2576cc)

**Comment by summaryInfo** [comment]
Hello!
Can you please provide config file if it exists and logs for the terminal started as `nsst --trace-events --trace-input --trace-controls --trace-characters` for the same sequence of inputs? Also, are you using X11 or Wayland?

**Comment by HalanoSiblee** [comment]
`nsst --trace-events --trace-input --trace-controls --trace-characters nvim` As I entered insert mode and use numpad
```
[INFO] Seq: ^[[K
[INFO] Seq: ^[(B
[INFO] Seq: ^[[m
[INFO] Seq: ^[[38;2;201;199;190m
[INFO] Seq: ^M
[INFO] Seq: ^J
[INFO] Seq: ^[[J
[INFO] Seq: ^[[1;5H
[INFO] Seq: ^[[?25h
[INFO] Seq: ^[[?25l
[INFO] Seq: ^[[?1004h
[INFO] Rep: ^[[I	
[INFO] Seq: ^[[?25h
[INFO] Event: event=KeyPress win=0x1800004 keycode=0x1f state=10
[INFO] Key: sym=0x69 mask=0x10 ascii=0x69 utf32=0x69
[INFO] Key char: (69) 'i' 
[INFO] Seq: ^[[?25l
[INFO] Seq: ^[[24;70H
[INFO] Char: (69) 'i' 
[INFO] Seq: ^[[1;5H
[INFO] Seq: ^[[?25h
[INFO] Seq: ^[[?25l
[INFO] Seq: ^[[24;70H
[INFO] Char: (20) ' ' 
[INFO] Seq: ^M
[INFO] Seq: ^[[A
[INFO] Seq: ^[(B
[INFO] Seq: ^[[0;1m
[INFO] Seq: ^[[38;2;198;92;194m
[INFO] Seq: ^[[48;2;41;41;41m
[INFO] Char: (20) ' ' 
[INFO] Char: (e7c5) '' 
[INFO] Seq: ^[[23;3H
[INFO] Char: (20) ' ' 
[INFO] Char: (49) 'I' 
[INFO] Char: (4e) 'N' 
[INFO] Char: (53) 'S' 
[INFO] Char: (45) 'E' 
[INFO] Char: (52) 'R' 
[INFO] Char: (54) 'T' 
[INFO] Char: (20) ' ' 
[INFO] Seq: ^[(B
[INFO] Seq: ^[[m
[INFO] Seq: ^[[38;2;102;102;102m
[INFO] Char: (20) ' ' 
[INFO] Char: (5021a) '󰈚' 
[INFO] Seq: ^[[23;13H
[INFO] Char: (20) ' ' 
[INFO] Char: (45) 'E' 
[INFO] Char: (6d) 'm' 
[INFO] Char: (70) 'p' 
[INFO] Char: (74) 't' 
[INFO] Char: (79) 'y' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (20) ' ' 
[INFO] Char: (4c) 'L' 
[INFO] Char: (6e) 'n' 
[INFO] Char: (20) ' ' 
[INFO] Char: (30) '0' 
[INFO] Char: (2c) ',' 
[INFO] Char: (20) ' ' 
[INFO] Char: (43) 'C' 
[INFO] Char: (6f) 'o' 
[INFO] Char: (6c) 'l' 
[INFO] Char: (20) ' ' 
[INFO] Char: (31) '1' 
[INFO] Seq: ^G
[INFO] Seq: ^[[6 q
[INFO] Seq: ^[[1;5H
[INFO] Seq: ^[[?25h
[INFO] Event: event=KeyPress win=0x1800004 keycode=0x3e state=10
[INFO] Key: sym=0xFFE2 mask=0x10 ascii=0x0 utf32=0x0
[INFO] Event: XKB Event 2
[INFO] Event: event=KeyPress win=0x1800004 keycode=0x26 state=11
[INFO] Key: sym=0x41 mask=0x11 ascii=0x41 utf32=0x41
[INFO] Key seq: ^[[27;2;65~
[INFO] Seq: ^[[?25l
[INFO] Seq: ^[(B
[INFO] Seq: ^[[m
[INFO] Seq: ^[[38;2;201;199;190m
[INFO] Char: (41) 'A' 
[INFO] Seq: ^[[23;72H
[INFO] Seq: ^[(B
[INFO] Seq: ^[[m
[INFO] Seq: ^[[38;2;102;102;102m
[INFO] Char: (31) '1' 
[INFO] Char: (2c) ',' 
[INFO] Char: (20) ' ' 
[INFO] Char: (43) 'C' 
[INFO] Char: (6f) 'o' 
[INFO] Char: (6c) 'l' 
[INFO] Char: (20) ' ' 
[INFO] Char: (32) '2' 
[INFO] Seq: ^[[1;6H
[INFO] Seq: ^[[?25h
[INFO] Event: XKB Event 2
[INFO] Event: event=KeyPress win=0x1800004 keycode=0x57 state=10
[INFO] Key: sym=0xFFB1 mask=0x10 ascii=0x0 utf32=0x31
[INFO] Key seq: ^[Oq
[INFO] Event: event=KeyPress win=0x1800004 keycode=0x58 state=10
[INFO] Key: sym=0xFFB2 mask=0x10 ascii=0x0 utf32=0x32
[INFO] Key seq: ^[Or
[INFO] Event: event=KeyPress win=0x1800004 keycode=0x59 state=10
[INFO] Key: sym=0xFFB3 mask=0x10 ascii=0x0 utf32=0x33
[INFO] Key seq: ^[Os
[INFO] Event: event=KeyPress win=0x1800004 keycode=0x53 state=10
[INFO] Key: sym=0xFFB4 mask=0x10 ascii=0x0 utf32=0x34
[INFO] Key seq: ^[Ot
[INFO] Seq: ^[[?25l
[INFO] Seq: ^[[23;79H
[INFO] Char: (31) '1' 
[INFO] Seq: ^[[1;5H
[INFO] Seq: ^[[?25h
[INFO] Event: event=KeyPress win=0x1800004 keycode=0x54 state=10
[INFO] Key: sym=0xFFB5 mask=0x10 ascii=0x0 utf32=0x35
[INFO] Key seq: ^[Ou
[INFO] Seq: ^[[?25l
[INFO] Seq: ^[(B
[INFO] Seq: ^[[m
[INFO] Seq: ^[[38;2;201;199;190m
[INFO] Char: (3c) '<' 
[INFO] Char: (6b) 'k' 
[INFO] Char: (4f) 'O' 
[INFO] Char: (72) 'r' 
[INFO] Char: (69) 'i' 
[INFO] Char: (67) 'g' 
[INFO] Char: (69) 'i' 
[INFO] Char: (6e) 'n' 
[INFO] Char: (3e) '>' 
[INFO] Char: (41) 'A' 
[INFO] Seq: ^[[23;68H
[INFO] Seq: ^[(B
[INFO] Seq: ^[[m
[INFO] Seq: ^[[38;2;102;102;102m
[INFO] Char: (4c) 'L' 
[INFO] Char: (6e) 'n' 
[INFO] Char: (20) ' ' 
[INFO] Char: (31) '1' 
[INFO] Char: (2c) ',' 
[INFO] Char: (20) ' ' 
[INFO] Char: (43) 'C' 
[INFO] Char: (6f) 'o' 
[INFO] Char: (6c) 'l' 
[INFO] Char: (20) ' ' 
[INFO] Char: (31) '1' 
[INFO] Char: (30) '0' 
[INFO] Seq: ^[[1;14H
[INFO] Seq: ^[[?25h
[INFO] Event: event=KeyPress win=0x1800004 keycode=0x55 state=10
[INFO] Key: sym=0xFFB6 mask=0x10 ascii=0x0 utf32=0x36
[INFO] Key seq: ^[Ov
[INFO] Seq: ^[[?25l
[INFO] Seq: ^[[23;79H
[INFO] Char: (31) '1' 
[INFO] Seq: ^[[1;15H
[INFO] Seq: ^[[?25h
[INFO] Event: event=KeyPress win=0x1800004 keycode=0x25 state=10
[INFO] Key: sym=0xFFE3 mask=0x10 ascii=0x0 utf32=0x0
[INFO] Event: XKB Event 2
[INFO] Event: event=KeyPress win=0x1800004 keycode=0x36 state=14
[INFO] Key: sym=0x63 mask=0x14 ascii=0x63 utf32=0x3
[INFO] Key seq: ^[[27;5;99~
[INFO] Seq: ^[[?25l
[INFO] Seq: ^M
[INFO] Seq: ^[[22B
[INFO] Seq: ^[(B
[INFO] Seq: ^[[0;1m
[INFO] Seq: ^[[38;2;37;176;188m
[INFO] Seq: ^[[48;2;41;41;41m
[INFO] Char: (20) ' ' 
[INFO] Char: (e7c5) '' 
[INFO] Seq: ^[[23;3H
[INFO] Char: (20) ' ' 
[INFO] Char: (4e) 'N' 
[INFO] Char: (4f) 'O' 
[INFO] Char: (52) 'R' 
[INFO] Char: (4d) 'M' 
[INFO] Char: (41) 'A' 
[INFO] Char: (4c) 'L' 
[INFO] Char: (20) ' ' 
[INFO] Seq: ^[(B
[INFO] Seq: ^[[m
[INFO] Seq: ^[[38;2;102;102;102m
[INFO] Char: (20) ' ' 
```
Last configs :
```ini
alpha=0.5
font=BlexMono Nerd Font Mono-11

font-gamma=1.2
pixel-mode=mono
override-boxdrawing=true
erase-scrollback=false
visual-bell=false
bell=off

background=#000000
foreground=#FFFFFF
cursor-background=#000000
cursor-foreground=#FFFFFF

color0=#1c2023
color1=#c7ae95
color2=#95c7ae
color3=#aec795
color4=#ae95c7
color5=#c795ae
color6=#95aec7
color7=#c7ccd1

color8=#747c84
color9=#c7ae95
color10=#95c7ae
color11=#aec795
color12=#ae95c7
color13=#c795ae
color14=#95aec7
color15=#f3f4f5

key-break=Break
key-numlock=T-Num_Lock
key-scroll-up=C-Up
key-scroll-down=C-Down
key-inc-font=T-Page_Up
key-dec-font=T-Page_Down
key-reset-font=T-Home
key-new-window=T-N
key-reset=T-R
key-reload-config=T-X
key-reverse-video=T-I
key-copy=T-C
key-copy-uri=T-U
key-paste=T-V
key-jump-next-cmd=T-F
key-jump-prev-cmd=T-B
```
on wayland I use foot terminal didn't try nsst their
on X11 I started use nsst because it has the server and client feature So yeah It ran on X11 system.

**Comment by summaryInfo** [comment]
Thanks for the feedback!
Has the issue been fixed for you in the master by 2186063?

**Comment by summaryInfo** [comment]
Feel free to submit feature requests, I'll take a look in my free time.
